src/manager/k8s.service.ts

```diff
@@ -1,24 +1,193 @@
 import { Injectable } from '@nestjs/common';
 import { IAgentManager } from './agent.manager.interface';
 import { AgentConfig } from './agent.config';
+import { KubeConfig, CoreV1Api, V1PodCondition } from '@kubernetes/client-node';
+import { readFile } from 'fs';
+import cryptoRandomString from 'crypto-random-string';
 
 /**
  * Starts and stops agents within a kubernetes environment
  */
 @Injectable()
 export class K8sService implements IAgentManager {
 
+    private kapi : CoreV1Api = this.newKubeClient();
+    private namespace : string = this.getNamespace();
+
+    // newKubeClient
+    // This configuration is only available if you run the server on a filesystem provisioned with a k8s client.
+    // When running in k8s, this config will be available via the configured k8s service account of the pod.
+    // Locally, this config is provided via ~/.kube/config.
+    private newKubeClient(): CoreV1Api {
+        const kc = new KubeConfig();
+        kc.loadFromDefault();
+        const k = kc.makeApiClient(CoreV1Api);
+        return k;
+    }
+
+    // getNamespace
+    // This will work in k8s when the service account is set up correctly.
+    // Locally this will fail, unless you create this file at the correct path.
+    private getNamespace(): string {
+      let namespace = 'default';
+      readFile('/var/run/secrets/kubernetes.io/serviceaccount/namespace','utf8', (err,data) => {
+        if (err) throw err;
+        namespace = data;
+      });
+      return namespace;
+    }
+
+    private async createPod(config: AgentConfig, id: string): Promise<any> {
+      const inboundTransportSplit = config.inboundTransport.split(' ');
+      const adminSplit = config.admin.split(' ');
+      const port: any = 3010;
+      const res = await this.kapi.createNamespacedPod(this.namespace, {
+        apiVersion: 'v1',
+        kind: 'Pod',
+        metadata: {
+          name: `agent-${id}`,
+          labels: {
+            'app.kubernetes.io/instance': `agent-${id}`,
+            'app.kubernetes.io/name': `agent-${id}`
+          }
+        },
+        spec: {
+          containers: [{
+            name: `agent-${id}`,
```

Comment:
I'm not sure of the syntax for k8s, but in docker, the container has a name, which we can then use for networking calls, eg http://container-name - so we may want to change this to config.label like in https://github.com/kiva/aries-guardianship-agency/blob/master/src/manager/docker.service.ts#L33